### PR TITLE
 [interoperator] Dynamic Watches 

### PR DESCRIPTION
**Pull request by vivekzhere**
- Setup watches reading a config map
- Updated binding and instance controllers
- Watches set only on startup
- Add sample config map
- Populate configmap on startup
- Remove use go client-go directly
- Use sources template to compute watches
- Restart interoperator on change in watch list

### CODE REVIEW

**Comment by vivekzhere** [comment]
Tested On K8s. Few more unit tests need to be added. Working on it. Ready for Review

**Comment by vivekzhere** [comment]
@amitmalav @subhankarc @jayeshjk Ready for review

File: interoperator/pkg/controller/sfplan/sfplan_controller.go [hunk 1]
```diff
@@ -65,18 +74,39 @@ func add(mgr manager.Manager, r reconcile.Reconciler) error {
 	return nil
 }
 
+func restartOnWatchUpdate(mgr manager.Manager, initWatches chan bool) {
+	for {
+		select {
+		case <-initWatches:
```
Comment (amshuman-kr) [inline interoperator/pkg/controller/sfplan/sfplan_controller.go]: Shouldn't we select from `stopCh` as well to handle the graceful termination?

File: interoperator/config/samples/interoperator-config.yaml [hunk 2]
```diff
@@ -0,0 +1,35 @@
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: interoperator-config
+data:
+  instanceWorkerCount: "5"
```
Comment (amshuman-kr) [inline interoperator/config/samples/interoperator-config.yaml]: Anyway we have to parse YAML for the deeper structs for `instanceContollerWatchList` and `bindingContollerWatchList`. Why not just bite the bullet and dump the whole YAML as a single key in the configmap?

File: interoperator/pkg/internal/config/config.go [hunk 3]
```diff
@@ -98,16 +118,102 @@ func (cfg *config) GetConfig() *InteroperatorConfig {
...
+	toCreate := false
+	if apiErrors.IsNotFound(err) {
+		toCreate = true
+		cfg.configMap = &corev1.ConfigMap{}
+		cfg.configMap.SetName(constants.ConfigMapName)
+		cfg.configMap.SetNamespace(cfg.namespace)
+		cfg.configMap.Data = make(map[string]string)
+	}
+
+	val := reflect.ValueOf(interoperatorConfig)
```
Comment (amshuman-kr) [inline interoperator/pkg/internal/config/config.go]: If we dump the whole yaml as a single key/value in the `ConfigMap` as mentioned [above](https://github.com/cloudfoundry-incubator/service-fabrik-broker/pull/622/files#diff-5c71366ff7ef7d1a65501881ecb6544cR7), we can then simply use YAML/JSON (en|de)coder on the whole thing instead the current mixture of `reflect` and (en|de)coding.

File: interoperator/pkg/controller/sfplan/sfplan_controller.go (hunk 1 above)
Comment (vivekzhere) [reply interoperator/pkg/controller/sfplan/sfplan_controller.go]: Added

File: interoperator/config/samples/interoperator-config.yaml (hunk 2 above)
Comment (vivekzhere) [reply interoperator/config/samples/interoperator-config.yaml]: Updated.

File: interoperator/pkg/internal/config/config.go (hunk 3 above)
Comment (vivekzhere) [reply interoperator/pkg/internal/config/config.go]: Use of reflect removed

**Pull request merged**